internal/service/advancedclustertpf/resource.go

```diff
@@ -344,6 +350,19 @@ func (r *rs) updateAndWaitLegacy(ctx context.Context, patchReq *admin20240805.Cl
 	return AwaitChanges(ctx, r.Client.AtlasV2.ClustersApi, &tfModel.Timeouts, diags, projectID, clusterName, changeReasonUpdate)
 }
 
+func (r *rs) applyTenantUpgrade(ctx context.Context, plan *TFModel, upgradeRequest *admin.LegacyAtlasTenantClusterUpgradeRequest, diags *diag.Diagnostics) *admin.ClusterDescription20240805 {
+	api := r.Client.AtlasV2.ClustersApi
```

Comment:
nit: for consistency on how we usually use the SDK
```
connV2 := r.Client.AtlasV2
...
connV2.ClustersApi.UpgradeSharedCluster(ctx, projectID, upgradeRequest).Execute()
```

Reply:
This file differs, using `api` everywhere, and being intentional about the version